modules/ingest-user-agent/src/main/java/org/elasticsearch/ingest/useragent/UserAgentProcessor.java

```diff
@@ -272,4 +260,11 @@ public static Property parseProperty(String propertyName) {
             }
         }
     }
+
+    static boolean maybeUpgradeConfig(Map<String, Object> config) {
```

Comment:
Question: Are we expected to support upgrades which skip a major version? If not, I think we can marked this for removal in v10. This is because it's not possible to add processors with the deprecated parameter in v9, and any added in v8 or earlier will have been fixed by the upgrade to v9.

Reply:
All upgrades to vMAJOR have to go through the last minor of the previous major (v[MAJOR-1].LAST)